### ISSUE TITLE
Add OO wrapper for [tcl-function]

### ISSUE THREAD

**Issue by rkeene** [dbohdan/tcl-duktape#8]
Before v0.7.0: Add oo.tcl wrapper for `tcl-function`

**Comment by dbohdan** [comment]
While it would be nice, this is optional.  I won't require the OO wrapper to fully reflect the procedural API.  The gap would be easier to work around if the OO wrapper had a method to return the internal `id` of the JS interpreter.

**Comment by dbohdan** [comment]
>The gap would be easier to work around if the OO wrapper had a method to return the internal `id` of the JS interpreter.

Done. We don't care about encapsulation here.

**Comment by hvxl** [comment]
This is the first thing I also missed within 5 minutes of playing with the library. While I can understand that the OO wrapper may not fully reflect the procedural API, the task done by tcl-function is a major cornerstone of js integration. So I would also like it to be added to the OO wrapper.

For now I will just use the following sequence:
```
package require duktape::oo

oo::define ::duktape::oo::Duktape {
    method tcl-function args {
        my variable id
        ::duktape::tcl-function $id {*}$args
    }
}
```

**Comment by hvxl** [comment]
In light of #11, make that:
```
oo::define ::duktape::oo::Duktape {
    method tcl-function args {
        my variable id
        uplevel 1 [list ::duktape::tcl-function $id {*}$args]
    }
}
```

**Comment by dbohdan** [comment]
Fixed via #14.
